src/main/java/com/indeed/jiraactions/ActionFactory.java

```diff
@@ -39,6 +47,8 @@ public Action create(final Issue issue) throws IOException {
         final User assignee = userLookupService.getUser(issue.initialValueKey("assignee", "assigneekey"));
         final User reporter = userLookupService.getUser(issue.initialValueKey("reporter", "reporterkey"));
         final User creator = issue.fields.creator == null ? User.INVALID_USER : userLookupService.getUser(issue.fields.creator.getKey());
+        final List<Component> components = null == issue.fields.components ? emptyList() : ImmutableList.copyOf(issue.fields.components);
```

Comment:
There are two ways we get the value of a field at creation time. The first is when the issue is created with a given field having a specific value, and it never changes. This will pick that up. The second way is when we see a field modified in the history. This won't pick that up.

Reproduction case:
* Create an issue with a specific component
* Modify the issue to remove the component

In that case, `issue.fields.components` will be empty, but there will be a history item like:

`{
"field": "Component",
"fieldtype": "jira",
"from": "<someId>",
"fromString": "<someComponentName>",
"to": null,
"toString": null
}`